README.md

```diff
@@ -1,39 +1,134 @@
+The Compliance Framework Agent, is the central component responsible for running plugins on schedule, passing them
 policies, and keeping plugins and policies up to date based on upstream plugins and policies.
 
 ## Plugins
 
-Plugins are the primary method of gathering data for policy checks. Plugins will execute some code to fetch the 
+Plugins are the primary method of gathering data for policy checks. Plugins will execute some code to fetch the
 data necessary for specific compliance checks, and then run these against policies to ensure compliance with a
-businesses' policies. 
+businesses' policies.
 
 As an example, there is a plugin called `local-ssh`. This plugin will retrieve the SSH configuration on a machine,
-convert it to a usable json structure, and then run company policies against the SSH configuration to ensure a host 
+convert it to a usable json structure, and then run company policies against the SSH configuration to ensure a host
 machine complies with all regulatory and security policies.
 
 The Agent is responsible for starting and calling these plugins, when it is necessary, usually on a set schedule.
 
-Plugins are entirely flexible in that they can be used to test any type of configuration or data, as long as they report 
-findings and observations about what they found. 
+Plugins are entirely flexible in that they can be used to test any type of configuration or data, as long as they report
+findings and observations about what they found.
 
 ## Policies
 
-Polices, although not strictly required, are written in Rego, and passed to each plugin so it can assert whether 
-the data it has collected, conforms with organisational policies. 
+Polices, although not strictly required, are written in Rego, and passed to each plugin so it can assert whether
+the data it has collected, conforms with organisational policies.
 
 For each violation of the policies, the plugin will report findings and observations to the agent, which in turn will
 report these to the central configuration api.
 
 ## NATS
 
-Upon the plugin(s) running, they'll send the results of the Observations and/or Findings to an event queue (-flag configured).
+Upon the plugin(s) running, they'll send the results of the Observations and/or Findings to an event queue (-flag
+configured).
 
-To run an instance of NATS, checkout the [local-dev](https://github.com/compliance-framework/local-dev) repository: 
+To run an instance of NATS, checkout the [local-dev](https://github.com/compliance-framework/local-dev) repository:
+
+## Configuration
+
+The agent must be configured using a configuration file that can be in any of YAML, JSON or TOML. We'll assume YAML
+because it's fairly human readable and wide spread.
+
+The basic configuration file looks something like the following fields:
+
+```
+nats:
+  domain: <nats_domain>
+  port: <nats_port>
+
+plugins:
+  <plugin_identifier>:  # Can have as many of these as you like
+    assessment_plan_id: <assessment_plan_id>
```

Comment:
I believe we need to specify multiple here to cater for when a check is feeding into multiple assessment plans. 
As a concrete example, we could have 2 assessment plans:
* Ensure SSH Security of publicly exposed hosts
* Ensure SSH Security of production Kubernetes cluster
As part of both assessment plans, we need to check the SSH configuration on the same machine. 
In this case we do the check once, and report the result for both assessment plans.